activerecord/lib/active_record/relation/predicate_builder/association_query_value.rb

```diff
@@ -31,9 +31,8 @@ def primary_key
         end
 
         def convert_to_id(value)
-          case value
-          when Base
```

Comment:
This `case` was originally introduced in https://github.com/rails/rails/commit/cd0ed12d1a9edc73be953d700748b5827df890c7 by @sgrif. 

Do you recall if there was a reason that `case/when` was used instead of `#respond_to?(primary_key)`?